### DISCUSSION TITLE
Route model binding supported?

### DISCUSSION THREAD

**Discussion by ericp-mrel** [wintercms/winter#717] [Q&A] [answered]
Hi,

I'm trying to use route model binding according to the Laravel documentation, but it is not working no matter what I do. https://laravel.com/docs/9.x/routing#customizing-the-key

I've tried doing something like this:
URL: `https://example.test/pdf/acme/my-slug/details` and `https://example.test/pdf/acme/1/details`
but I just get a new model instance, and it doesn't throw an exception if it can't find a record with the id/slug either.
```php
Route::get('/pdf/acme/{foo}/details', function (Foo $foo) {
// ...
});
```
and

```php
Route::get('/pdf/acme/{foo:slug}/details', function (Foo $foo) {
// ...
});
```

Just wondering if this is supported by Winter CMS? I assume it should be since this is just standard Laravel routing.

**Comment by bennothommo** [comment]
We definitely don't override much with the Routing side of things, but I believe we do handle dependency injection a little differently, which might be why this is not working.

I'll convert this to an issue. (https://github.com/wintercms/winter/issues/718)

**Comment by mjauvin** [comment, marked as answer]
@ericp-mrel I found the issue... in order for the implicit model binding to work, you need to use the "web" group middleware in your route:
```
Route::get('/pdf/acme/{foo}/details', function (Foo $foo) {
    // ...
})->middleware('web');
```

Alternatively, you can use the `\Illuminate\Routing\Middleware\SubstituteBindings` middleware directly:

```
Route::get('/pdf/acme/{foo}/details', function (Foo $foo) {
    // ...
})->middleware('\Illuminate\Routing\Middleware\SubstituteBindings');

**Reply by ericp-mrel** [reply]
Ahhh, totally forgot about adding the web middleware.

Thank you
